src/actors/objects/mod.rs

```diff
@@ -585,23 +581,22 @@ impl ObjectsActor {
     }
 }
 
-fn decompress_object_file(
-    cache_key: &CacheKey,
-    download_file_path: &Path,
-    mut download_file: fs::File,
-    mut extract_file: fs::File,
-) -> io::Result<fs::File> {
+/// Decompresses an object file.
+///
+/// Some compression methods are implemented by spawning an external tool and can only
+/// process from a named pathname, hence we need a [NamedTemporaryFile] as source.
+fn decompress_object_file(src: &NamedTempFile, mut dst: fs::File) -> io::Result<fs::File> {
     // Ensure that both meta data and file contents are available to the
     // subsequent reads of the file metadata and reads from other threads.
-    download_file.sync_all()?;
+    src.as_file().sync_all()?;
```

Comment:
maybe introduce a `src_file` local instead of doing a call all the time?

Reply:
heh, i did that first but ended up not liking it.  It read a bit weird with two same-named variables but one of them being in hungarian notation.  This is more explicit and unambigiuous.